### DISCUSSION TITLE
Disable a single flash message in flask_security

### DISCUSSION THREAD

**Discussion by bkiziuk** [pallets-eco/flask-security#968] [General]
Before flask_security_too 5.4.0, I used the:
`SECURITY_MSG_LOGIN = ('', '')`
in order to disable this single flash message.
After upgrading to 5.4.3, the `SECURITY_MSG_LOGIN` was changed to `SECURITY_MSG_UNAUTHENTICATED`.
Unfortunately, if I do:
`SECURITY_MSG_UNAUTHENTICATED = ('', '')`
the flash message isn't disabled, but instead shows:
```
Project-Id-Version: PROJECT VERSION Report-Msgid-Bugs-To: EMAIL@ADDRESS POT-Creation-Date: 2023-04-16 17:50+0200 PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE Last-Translator: FULL NAME <EMAIL@ADDRESS> Language: pl Language-Team: pl <LL@li.org> Plural-Forms: nplurals=3; plural=(n==1 ? 0 : n%10>=2 && n%10<=4 && (n%100<10 || n%100>=20) ? 1 : 2); MIME-Version: 1.0 Content-Type: text/plain; charset=utf-8 Content-Transfer-Encoding: 8bit Generated-By: Babel 2.11.0
```
which is probably because the .po file at its beginning includes:
```
# Polish translations for PROJECT.
# Copyright (C) 2023 ORGANIZATION
# This file is distributed under the same license as the PROJECT project.
# FIRST AUTHOR <EMAIL@ADDRESS>, 2023.
#
msgid ""
msgstr ""
"Project-Id-Version: PROJECT VERSION\n"
"Report-Msgid-Bugs-To: EMAIL@ADDRESS\n"
"POT-Creation-Date: 2023-02-20 10:39+0100\n"
"PO-Revision-Date: 2023-02-19 16:04+0100\n"
"Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
"Language: pl\n"
"Language-Team: pl <LL@li.org>\n"
"Plural-Forms: nplurals=3; plural=(n==1 ? 0 : n%10>=2 && n%10<=4 && "
"(n%100<10 || n%100>=20) ? 1 : 2);\n"
"MIME-Version: 1.0\n"
"Content-Type: text/plain; charset=utf-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Generated-By: Babel 2.11.0\n"
```
and the empty string is substituted with the rest of the text. The .po file is inline with the intended one according to Babel documentation.
Is there any simple way to overcome it? I just need to get rid of a `SECURITY_MSG_UNAUTHENTICATED` but not the rest.

**Comment by jwag956** [comment]
Did you try the suggestions listed in: https://flask-security-too.readthedocs.io/en/stable/customizing.html#overriding-messages

**Reply by bkiziuk** [reply]
Oh, yes.
As the docs say:

> The string messages themselves are a ‘key’ into the translation .po/.mo files.

I can override the message with non-empty one. What I cannot do is to replace it with empty message as then the empty message is substituted with the `msgstr""` which in the compiled .mo file is then followed by PO info.
But after a little more digging I can see that this is not flask_security_too issue but Babel issue and there's a ticket opened for it back in 2015 [https://github.com/python-babel/babel/issues/157](https://github.com/python-babel/babel/issues/157)
Not really sure why the previous version worked, some kind of regression maybe.
